pathfinder/tests/fixtures/core.py

```diff
@@ -1,42 +1,47 @@
 import random
 from typing import List
 from unittest.mock import Mock
-
 import pytest
-from eth_utils import to_checksum_address, encode_hex, keccak
+from coincurve import PrivateKey
+from eth_utils import to_checksum_address, remove_0x_prefix
 from raiden_libs.contracts import ContractManager
+from raiden_libs.utils import public_key_to_address
 from web3 import Web3
 from web3.contract import Contract
-
 from pathfinder.pathfinding_service import PathfindingService
+from pathfinder.tests.config import NUMBER_OF_CHANNELS
 from pathfinder.token_network import TokenNetwork
+from pathfinder.utils.types import Address
 
 
 @pytest.fixture
 def populate_token_networks(
         token_networks: List[TokenNetwork],
         token_network_contracts: List[Contract],
+        addresses: List[Address],
+        private_key_seeds: List[str],
 ) -> None:
-
+    random.seed(NUMBER_OF_CHANNELS)
     for token_network in token_networks:
         # some random magic to add channels
-        for channel_id in range(100):
-            seed1, seed2 = random.sample(range(20), 2)
-
-            p1 = to_checksum_address(encode_hex(keccak(seed1))[:42])
-            p2 = to_checksum_address(encode_hex(keccak(seed2))[:42])
-
+        for channel_id in range(NUMBER_OF_CHANNELS):
+            seed1, seed2 = random.sample(private_key_seeds, 2)
+            private_key1 = PrivateKey.from_hex(remove_0x_prefix(seed1))
+            private_key2 = PrivateKey.from_hex(remove_0x_prefix(seed2))
+            public_key1 = to_checksum_address(public_key_to_address(private_key1.public_key))
+            public_key2 = to_checksum_address(public_key_to_address(private_key2.public_key))
```

Comment:
These are addresses, not public keys. Ethereum doesn't have a different definition of public key as it does with its private key counterpart. An Ethereum public key is the ECDSA public key, which again is different from the address. You cannot even reconstruct the public key from an address. See the definition of `public_key_to_address` for more information.

Renaming suggestions here:
* `seed` to `private_key`
* `public_key` to `address`
* `private_key` to `private_key_ecdsa` or something, to differentiate it from the Ethereum private key

Reply:
Agreed on 2 and 3!